service/storage_proxy.cc

```diff
@@ -1550,9 +1550,25 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
         return _mutation_holder->store_hint(hm, ep, tr_state);
     }
     future<> apply_locally(storage_proxy::clock_type::time_point timeout, tracing::trace_state_ptr tr_state) {
+        // Writes to local tables don't have to be fenced. Moreover, they shouldn't, as they can cause
+        // errors on the bootstrapping node. A specific scenario:
+        // 1. A write request on a bootstrapping node is initiated. The write handler captures the
+        // effective replication map. Let's suppose its version is 5.
+        // 2. The handler is yielded on some co_await.
+        // 3. The topology coordinator is progressing, and it doesn't wait for pending requests on this
+        // node because it's not in the list of normal nodes.
+        // 4. The topology coordinator increments the fencing version to 6 and writes it to the RAFT table.
+        // 5. The bootstrapping node processes this Raft log entry. The fence version is updated to 6.
+        // 6. The initial write handler wakes up. It sees that 6 > 5 and fails.
+        // This scenario is only possible for local writes. Non-local writes never happen on the
+        // bootstrapping node because it hasn't been published to clients yet.
+        auto& keyspace = _proxy->local_db().find_keyspace(_mutation_holder->schema()->ks_name());
+        auto replication_strategy_type = keyspace.get_replication_strategy().get_type();
+        auto fence = replication_strategy_type == locator::replication_strategy_type::local ?
+                fencing_token{} : fencing_token{_effective_replication_map_ptr->get_token_metadata().get_version()};
```

Comment:
we can use `_effective_replication_map_ptr->get_replication_strategy()->get_type()`

And probably it's better to extract `get_fence` (e.g. as `storage_proxy::get_fence(erm)`), reuse it in both bases and move the comment inside it.

Reply:
> we can use `_effective_replication_map_ptr->get_replication_strategy()->get_type()`


Probably this suggestion prevents a measurable regression. It's also bad form to use an unfenced lookup (find_keyspace) in a fencing call.

Reply:
> we can use `_effective_replication_map_ptr->get_replication_strategy()->get_type()`
> 
> And probably it's better to extract `get_fence` (e.g. as `storage_proxy::get_fence(erm)`), reuse it in both bases and move the comment inside it.

I applied both suggestions.